instrumentation-core/spring/trace/src/main/java/io/opentelemetry/spring/instrumentation/trace/httpclients/RestTemplateInterceptor.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentelemetry.spring.instrumentation.trace.httpclients;
+
+import io.grpc.Context;
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+import java.io.IOException;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Wraps RestTemplate requests in a span. Adds the current span context to request headers.
+ *
+ */
+public class RestTemplateInterceptor implements ClientHttpRequestInterceptor {
+
+  private final static Logger LOG = LoggerFactory.getLogger(RestTemplateInterceptor.class.getName());
+
+  private final static HttpTextFormat.Setter<HttpRequest> SETTER =
```

Comment:
[Note] I feel like I've seen this before elsewhere. Should this be extracted as a separate class?

Reply:
I could try but the HttpTextFormat.Setter and HttpTextFormat.Getter implementations are heavily dependent on the web frameworks that are being implemented. There isn't a lot of overlap that can be easily extracted without increasing code complexity. I'd prefer to keep it this way for now